lib/weak_swagger_parameters/services/weak_parameters_options_adapter.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+module WeakSwaggerParameters
+  module Services
+    class WeakParametersOptionsAdapter
+      def self.adapt(options)
+        result = {}
+        result[:strong] = true
+        result[:required] = options[:required]
+        result[:only] = options[:enum] if options[:enum].present?
+        result = result.merge(range_options(options))
+
+        result
+      end
+
+      def self.range_options(options)
+        max = options[:max]
+        min = options[:min]
+        result = {}
+
+        if min.present?
+          max ||= MAX_VALUE
+        elsif max.present?
+          min ||= MIN_VALUE
+        end
+
+        result[:only] = min...max if min.present? && max.present?
+
+        result
+      end
```

Comment:
If they where any tests to this method I think it can be refactored to:

```ruby
      def self.range_options(options)
        return {} unless options.key?(:min) || options.key?(:max)

        max = options[:max] || MAX_VALUE
        min = options[:min] || MIN_VALUE
        { only: min...max }
      end
```